src/software/visualizer/widgets/visualizer.cpp

```diff
@@ -1,27 +1,79 @@
 #include "software/visualizer/widgets/visualizer.h"
 
-Visualizer::Visualizer() : QMainWindow(), main_widget(new MainWidget(this))
+#include "software/visualizer/geom/geometry_conversion.h"
+
+Visualizer::Visualizer(
+    std::shared_ptr<ThreadSafeBuffer<WorldDrawFunction>> world_draw_functions_buffer,
+    std::shared_ptr<ThreadSafeBuffer<AIDrawFunction>> ai_draw_functions_buffer,
+    std::shared_ptr<ThreadSafeBuffer<PlayInfo>> play_info_buffer,
+    std::shared_ptr<ThreadSafeBuffer<RobotStatus>> robot_status_buffer,
+    std::shared_ptr<ThreadSafeBuffer<Rectangle>> view_area_buffer)
+    : QMainWindow(),
+      main_widget(new MainWidget(this)),
+      update_timer(new QTimer(this)),
+      world_draw_functions_buffer(world_draw_functions_buffer),
+      ai_draw_functions_buffer(ai_draw_functions_buffer),
+      play_info_buffer(play_info_buffer),
+      robot_status_buffer(robot_status_buffer),
+      view_area_buffer(view_area_buffer),
+      update_timer_interval(Duration::fromSeconds(1.0 / 60.0))
 {
     setCentralWidget(main_widget);
+
+    connect(update_timer, &QTimer::timeout, this, &Visualizer::updateVisualizer);
+    update_timer->start(update_timer_interval.getMilliseconds());
 }
 
-Visualizer::~Visualizer()
```

Comment:
Why don't we need this destructor anymore?

Reply:
We actually don't need to manually destroy the raw pointers (Qt takes ownership), so have no need for the destructor